.github/ci_templates/helper_jobs.yml

```diff
@@ -86,109 +86,198 @@ git_info:
 {!{ define "pull_request_info_job" }!}
 # <template: pull_request_info>
 pull_request_info:
   name: Get pull request reference
   runs-on: ubuntu-latest
   outputs:
-    ref: ${{ steps.ref.outputs.ref }}
-    edition: ${{ steps.check_pr.outputs.edition || steps.check_push.outputs.edition }}
+    ref: ${{ steps.check_pr.outputs.ref }}
+    ref_slug: ${{ steps.check_pr.outputs.ref_slug }}
+    edition: ${{ steps.check_pr.outputs.edition }}
+    pr_title: ${{ steps.check_pr.outputs.pr_title }}
+    pr_description: ${{ steps.check_pr.outputs.pr_description }}
+    diff_url: ${{ steps.check_pr.outputs.diff_url }}
+    labels: ${{ steps.check_pr.outputs.labels }}
+
   # Skip pull_request and pull_request_target triggers for PRs authored by deckhouse-BOaTswain, e.g. changelog PRs.
   if: ${{ ! (startsWith(github.event_name, 'pull_request') && github.event.pull_request.user.login == 'deckhouse-BOaTswain') }}
   steps:
-    - name: Check push trigger
+    - name: Get PR info for push trigger
       id: check_push
       if: ${{ github.event_name == 'push' }}
       uses: {!{ index (ds "actions") "actions/github-script" }!}
       with:
         script: |
-          // Support 'push' trigger: use FE edition by default.
-          const edition = 'FE';
-          core.notice(`Enable default '${edition}' edition for 'push' trigger.`);
-          return core.setOutput('edition', edition);
+          // Support for 'push' trigger: find PR by commit SHA and pass response to check_pr step.
+          const { GITHUB_REF_NAME } = process.env
+          core.startGroup(`Fetching PR info for commit ${context.sha} in ${context.repo.name}:${GITHUB_REF_NAME} ...`)
+          try {
+            const response = await github.rest.repos.listPullRequestsAssociatedWithCommit({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                commit_sha: context.sha
+            });
+            if (response.status !== 200 || !response.data || response.data.length === 0) {
+              return core.setFailed(`Bad response on listing PRs for commit ${context.sha}: ${JSON.stringify(response)}`);
+            }
+            // Get first associated pr.
+            let pr = response.data[0];
+            core.info(`Current labels: ${JSON.stringify(pr.labels)}`);
+            // Reduce size to fit output limits.
+            pr = {
+              url:      pr.url,
+              diff_url: pr.diff_url,
+              number:   pr.number,
+              labels:   pr.labels,
+              head:     pr.head,
+              title:    pr.title,
+              body:     pr.body,
+            }
+            core.notice(`Found PR#{pr.number} for commit ${context.sha}`);
+            core.setOutput('pr_info', JSON.stringify(pr));
+          } catch (error) {
+            return core.setFailed(`Error listing pull requests for commit ${context.sha}: ${error}`)
+          } finally {
+            core.endGroup()
+          }
```

Comment:
JS in these two jobs (pull_request_info and git_info) is inevitable, because these jobs runs before checking for dangerous changes in PRs. We can transform these jobs to local actions as milestone-changelog, but its a future refactoring.